test: migrate path and paths to snapbox

eth3lbert opened the pull request:
### What does this PR try to resolve?

Part of https://github.com/rust-lang/cargo/issues/14039.

Migrate following to snapbox:

- `tests/testsuite/path.rs`
- `tests/testsuite/paths.rs`

eth3lbert commented on tests/testsuite/path.rs:
```diff
@@ -73,44 +72,49 @@ fn cargo_compile_with_nested_deps_shorthand() {
-        )
+        .with_stderr_data(str![[r#"
+[REMOVING] [ROOT]/foo/target
+[REMOVED] [..] files, [FILE_SIZE]B total
```
This highlights the potential benefit of redacting the number of files, as suggested in https://github.com/rust-lang/cargo/pull/14096#discussion_r1644832897.

weihanglo commented:
@bors r+

Thanks!

bors commented:
:pushpin: Commit 167f490731dcdbe9a51f5faa0776952b6e49bdc3 has been approved by `weihanglo`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/cargo) for this repository.

<!-- @bors r=weihanglo 167f490731dcdbe9a51f5faa0776952b6e49bdc3 -->
<!-- homu: {"type":"Approved","sha":"167f490731dcdbe9a51f5faa0776952b6e49bdc3","approver":"weihanglo","queue":"https://bors.rust-lang.org/homu/queue/cargo"} -->

bors commented:
:hourglass: Testing commit 167f490731dcdbe9a51f5faa0776952b6e49bdc3 with merge bc9b0d3eb825b3f1799316a50c2b4fad6d5fbccf...
<!-- homu: {"type":"BuildStarted","head_sha":"167f490731dcdbe9a51f5faa0776952b6e49bdc3","merge_sha":"bc9b0d3eb825b3f1799316a50c2b4fad6d5fbccf"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang/cargo/actions/runs/9605300584/job/26492846466)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang/cargo/actions/runs/9605300584/job/26492846466","builder_name":"checks-actions"} -->

weihanglo commented:
https://github.com/rust-lang/cargo/actions/runs/9605300584/job/26492637395#step:11:4033

You could add `edition = "2021"` to fix the test error.

eth3lbert commented:
> https://github.com/rust-lang/cargo/actions/runs/9605300584/job/26492637395#step:11:4033
> 
> You could add `edition = "2021"` to fix the test error.

Addressed and rebased! Thanks for the review!

eth3lbert commented:
Updated with auto-redaction applied!

weihanglo commented:
Thanks!

@bors r+

bors commented:
:pushpin: Commit c7926a128945802a8180f4e488d378c66e27310e has been approved by `weihanglo`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/cargo) for this repository.

<!-- @bors r=weihanglo c7926a128945802a8180f4e488d378c66e27310e -->
<!-- homu: {"type":"Approved","sha":"c7926a128945802a8180f4e488d378c66e27310e","approver":"weihanglo","queue":"https://bors.rust-lang.org/homu/queue/cargo"} -->

bors commented:
:hourglass: Testing commit c7926a128945802a8180f4e488d378c66e27310e with merge 9a7a603e5b1a5c2d4c38b0dacf31606e3ba29126...
<!-- homu: {"type":"BuildStarted","head_sha":"c7926a128945802a8180f4e488d378c66e27310e","merge_sha":"9a7a603e5b1a5c2d4c38b0dacf31606e3ba29126"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang/cargo/actions/runs/9636195564/job/26574166619)
Approved by: weihanglo
Pushing 9a7a603e5b1a5c2d4c38b0dacf31606e3ba29126 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"weihanglo","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang/cargo/actions/runs/9636195564/job/26574166619"},"merge_sha":"9a7a603e5b1a5c2d4c38b0dacf31606e3ba29126"} -->
